apps/aeoracle/src/aeo_extend_tx.erl

```diff
@@ -109,9 +109,10 @@ check(#oracle_extend_tx{nonce = Nonce, oracle_ttl = OTTL, fee = Fee} = Tx,
          fun() -> aetx_utils:check_account(OraclePK, Trees, Nonce, Fee) end,
          fun() -> ensure_oracle(OraclePK, Trees) end
          | case aetx_env:context(Env) of
-               aetx_contract -> []; %% TODO Cater for TTL fee from contract.
+               %% Contract is paying tx fee as gas.
+               aetx_contract -> [];
                aetx_transaction ->
-                   [fun() -> aeo_utils:check_ttl_fee(Height, OTTL, Fee) end]
+                   [fun() -> aeo_utils:check_ttl_fee(Height, OTTL, Fee) end] %% TODO Deduct portion of fee already accounted for by `aetx:check` before checking state TTL portion of fee.
```

Comment:
This (and the other introduced TODOs) needs `aetx:check` amendment first (it is WIP in one of @jur0 's branches?), correct?

Reply:
Yes. I will aim at getting #1796 merged before this one.